### PR TITLE
Support version 3 of Sendy API

### PR DESCRIPTION
**Pull request by mguinness**
See https://github.com/kloarubeek/SendyClient.Net/issues/8 for more details.

### CODE REVIEW

**Comment by mguinness** [comment]
Is there any way to find out what code needs to be refactored?  I'm not willing to submit additional permissions that the Better Code Hub is requesting.

File: src/Sendy.Client.csproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard1.6</TargetFramework>
     <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
-    <Version>2.2.0</Version>
+    <Version>2.3.0</Version>
```
Comment (kloarubeek) [inline src/Sendy.Client.csproj]: Please update it to 3.0.0 as it has breaking changes (by adding Async to all methods).

File: README.md [hunk 2]
```diff
@@ -87,9 +86,22 @@ var list = new MailingList
 list.CustomFields.Add(new CustomField("custom field 1"));
 list.CustomFields.Add(new CustomField("custom field 2", CustomField.DataTypes.Date));
 
-var result = await sendyClient.CreateList(list);
+var result = await sendyClient.CreateListAsync(list);
 ```
 
 
+## API updates
+The `SendyClient` class constructor now includes a new parameter named apiVer where you can pass in the version of Sendy you're working against.
```
Comment (kloarubeek) [inline README.md]: Good that you updated the readme with this 👏 
It's worth to mention why you need to pass on this version (to enable v3 functionality).

File: src/SendyClient.cs [hunk 3]
```diff
@@ -13,44 +13,58 @@ namespace Sendy.Client
 	public class SendyClient : IDisposable
 	{
 		private readonly string _apiKey;
-		private readonly HttpClient _httpClient;
+        private readonly Version _apiVer;
```
Comment (kloarubeek) [inline src/SendyClient.cs]: I see we're mixing tabs and spaces (sorry for the initial tab setup). Can you easily change your changes to use tabs? Or update everything to use spaces.

**Review by kloarubeek — CHANGES_REQUESTED** [review]
Thanks for the changes, looks good! If you change the minor changes I suggest, I can merge it and see what BetterCodeHub has to say.

**Comment by kloarubeek** [comment]
BetterCodeHub is complaining about this:

CreateCampaign and Subscribe are getting too big with too many parameters. You could fix this by creating a new model that holds all the parameters and moving some code from CreateCampaign to a private method.

image

**Comment by mguinness** [comment]
@kloarubeek Thanks for the feedback, I've been busy so I haven't gotten around to this yet.

**Comment by mguinness** [comment]
@kloarubeek Made changes as requested, let me know if any additional changes are needed.

**Review by kloarubeek — APPROVED** [review]

**Comment by kloarubeek** [comment]
Thanks for your help! I'll submit a brand new v3 to NuGet :+1:

**Comment by mguinness** [comment]
@kloarubeek Thanks Jereon!

**Pull request merged**